scanpy/tests/external/test_scrublet.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+import scanpy as sc
+import scanpy.external as sce
+
+pytest.importorskip("scrublet")
+
+
+def test_scrublet():
+    """
+    Test that Scrublet run works.
+
+    Check that scrublet runs and detects some doublets.
+    """
+    adata = sc.datasets.pbmc3k()
+    sce.pp.scrublet(adata)
+
+    errors = []
+
+    # replace assertions by conditions
+    if not set(['predicted_doublet', 'doublet_score']).issubset(adata.obs.columns):
+        errors.append("doublet columns not set in .obs")
+
+    if len(adata.obs_names[adata.obs.predicted_doublet]) == 0:
+        errors.append("no doublets identified")
+
+    assert not errors, "errors occured in scrublet testing:\n{}".format(
+        "\n".join(errors)
+    )
+
+
+def test_scrublet_simulate_doublets():
+    """
+    Test that standalone Scrublet doublet simulation works.
+
+    Check that doublet simulation runs and simulates some doublets..
+    """
+
+    adata_obs = sc.datasets.pbmc3k()
+    sc.pp.filter_genes(adata_obs, min_cells=3)
+    sc.pp.filter_cells(adata_obs, min_genes=3)
+    adata_obs.layers['raw'] = adata_obs.X
+    sc.pp.normalize_total(adata_obs)
+    logged = sc.pp.log1p(adata_obs, copy=True)
+
+    hvg = sc.pp.highly_variable_genes(logged)
+    adata_obs = adata_obs[:, logged.var['highly_variable']]
+
+    adata_sim = sce.pp.scrublet_simulate_doublets(adata_obs, raw_layer='raw')
```

Comment:
Didn't this `raw_layer` argument get changed?

How is this not causing test failures? I'm a bit worried these tests aren't being run.

Reply:
Argh- sorry. Would probably go through testing anyway, the rogue argument would be ignored and _get_obs_rep() would pick up .X. 

As mentioned above I couldn't work out why the tests aren't running here, was hoping you'd see it. I'll have a dig, but I'm clearly missing something about how this works so I'm not sure I'll be able to fix it by myself.

Reply:
Oh, sorry, totally missed that message.